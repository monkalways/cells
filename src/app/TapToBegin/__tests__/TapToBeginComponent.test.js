import React from 'react';
import { shallow } from 'enzyme';

import { TapToBeginComponent } from '../TapToBeginComponent';

describe('TapToBeginComponent', () => {
  let classes;
  let push;
  let first;
  let second;
  let versions;
  let getVersions;

  beforeEach(() => {
    classes = {
      body: 'body',
    };
    push = jest.fn();
    first = 'cells';
    second = 'c1';
    versions = [
      {
        name: 'Cell App',
        version: '1.0.0',
      },
      {
        name: 'Detainee Management Service',
        version: '1.0.0',
      },
      {
        name: 'EPROS Service',
        version: '1.0.0',
      },
    ];
    getVersions = jest.fn();
  });

  const setup = () => shallow(<TapToBeginComponent
    classes={classes}
    history={{ push }}
    match={{
      params: {
        first,
        second,
      },
    }}
    versions={versions}
    getVersions={getVersions}
  />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call get versions when componentDidMount is called', () => {
    const wrapper = setup();
    wrapper.instance().componentDidMount();

    expect(getVersions).toBeCalled();
  });

  it('should handle click', () => {
    const wrapper = setup();
    wrapper.find('#mainGrid').simulate('click');

    expect(push).toBeCalledWith(`/${first}/${second}/home`);
  });
});
