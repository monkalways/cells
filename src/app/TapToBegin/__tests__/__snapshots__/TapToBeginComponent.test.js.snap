// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TapToBeginComponent should render 1`] = `
<WithStyles(Grid)
  container={true}
  justify="center"
>
  <WithStyles(Grid)
    container={true}
    id="mainGrid"
    justify="center"
    onClick={[Function]}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(HeaderComponent)
        versions={
          Array [
            Object {
              "name": "Cell App",
              "version": "1.0.0",
            },
            Object {
              "name": "Detainee Management Service",
              "version": "1.0.0",
            },
            Object {
              "name": "EPROS Service",
              "version": "1.0.0",
            },
          ]
        }
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <img
        alt="EPS Logo"
        src="EPSlogo.png"
        width={350}
      />
      <WithStyles(Typography)
        align="center"
        variant="h4"
      >
        Tap to begin
      </WithStyles(Typography)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
